torch/csrc/jit/script/compiler.cpp

```diff
@@ -2937,17 +2937,14 @@ struct to_ir {
     // aten::slice, we should separate it from this function.
     if (dim) {
       AT_ASSERT(input->type()->isSubtypeOf(TensorType::get()));
-
       args.emplace_back(dim);
     } else {
       AT_ASSERT(!input->type()->isSubtypeOf(TensorType::get()));
     }
 
-    args.emplace_back(loc, "begin", emitExpr(Expr(slice.startOr(0))));
+    args.emplace_back(loc, "start", emitExpr(Expr(slice.startOr(INT64_MAX))));
```

Comment:
Can you just move the logic for default `start` and `end` within the TupleSlice? Because you've added defaults to the schema declaration in both list slicing and tensor slicing the default logic in the compiler is only necessary for the tuple slicing logic. There are also downstream uses of the indices in `lower_tuples.cpp` uses that (reasonably) assume that a start of 0 means that that the starting index is 0.  

It's a little unfortunate that we have a divergence between tuple slicing and tensor slicing, but that divergence already existed and I don't think it's worth reconciling.